lib/src/widgets/table_picker.dart

```diff
@@ -1 +1,276 @@
+  final Color dropDownColor;
+
+  ///[iconColor] to define the icon color
+  final Color iconColor;
+
+  ///[iconSize] to define the size of the icon
+  final double iconSize;
+
+  ///[padding] to define the padding of the dropdown
+  final EdgeInsetsGeometry padding;
+
+  ///[onOptionSelected] a callback function that returns the selected action from dropdown
+  final Function(EditTableEnum type) onOptionSelected;
+
+  @override
+  State<EditTableDropDown> createState() => _EditTableDropDownState();
+}
+
+class _EditTableDropDownState extends State<EditTableDropDown> {
+  late GlobalKey<ElTooltipState> _editTableETKey;
+
+  @override
+  void initState() {
+    _editTableETKey = GlobalKey<ElTooltipState>(
+        debugLabel: 'fontBgColorKey${widget.key.toString()}');
+    super.initState();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Padding(
+      padding: widget.padding,
+      child: SizedBox(
+        width: widget.iconSize,
+        height: widget.iconSize,
+        child: ElTooltip(
+          color: widget.dropDownColor,
+          distance: 0,
+          onTap: () {
+            if (MediaQuery.of(context).size.width < 480) {
+              _showEditTableSheet(true, context);
+            } else {
+              if (_editTableETKey.currentState != null) {
+                _editTableETKey.currentState!.showOverlayOnTap();
+              }
+            }
+          },
+          key: _editTableETKey,
+          content: _editTableListView(false, context),
+          child: SizedBox(
+            width: widget.iconSize,
+            height: widget.iconSize,
+            child: Image.asset(
+              ImageConstant.kiEditTablePng,
+              color: widget.iconColor,
+            ),
+          ),
+        ),
+      ),
+    );
+  }
+
+  Widget _getEditTableItem(
+      EditTableEnum type, BuildContext context, bool isMobile) {
+    String value = ""
+
+    switch (type) {
+      case EditTableEnum.insertRowAbove:
+        value = "Insert Row Above";
+        imagePath = ImageConstant.kiInsertTablePng;
+        break;
+      case EditTableEnum.insertRowBelow:
+        value = "Insert Row Below";
+        imagePath = ImageConstant.kiInsertRowBelowPng;
+        break;
+      case EditTableEnum.deleteRow:
+        value = "Delete Row";
+        imagePath = ImageConstant.kiDeleteRowPng;
+        break;
+      case EditTableEnum.deleteColumn:
+        value = "Delete Column";
+        imagePath = ImageConstant.kiDeleteColumnPng;
+        break;
+      case EditTableEnum.insertColumnLeft:
+        value = "Insert Column Left";
+        imagePath = ImageConstant.kiInsertColumnLeftPng;
+        break;
+      case EditTableEnum.insertColumnRight:
+        value = "Insert Column Right";
+        imagePath = ImageConstant.kiInsertColumnRightPng;
+        break;
+      case EditTableEnum.deleteTable:
+        value = "Delete Table";
+        imagePath = ImageConstant.kiDeleteTablePng;
+        break;
+    }
```

Comment:
**Suggestion:** Refactor the switch statement in the _getEditTableItem method to a map for better readability and performance.
```suggestion
    final Map<EditTableEnum, Map<String, String>> typeMap = {
      EditTableEnum.insertRowAbove: {
        'value': 'Insert Row Above',
        'imagePath': ImageConstant.kiInsertTablePng,
      },
      EditTableEnum.insertRowBelow: {
        'value': 'Insert Row Below',
        'imagePath': ImageConstant.kiInsertRowBelowPng,
      },
      EditTableEnum.deleteRow: {
        'value': 'Delete Row',
        'imagePath': ImageConstant.kiDeleteRowPng,
      },
      EditTableEnum.deleteColumn: {
        'value': 'Delete Column',
        'imagePath': ImageConstant.kiDeleteColumnPng,
      },
      EditTableEnum.insertColumnLeft: {
        'value': 'Insert Column Left',
        'imagePath': ImageConstant.kiInsertColumnLeftPng,
      },
      EditTableEnum.insertColumnRight: {
        'value': 'Insert Column Right',
        'imagePath': ImageConstant.kiInsertColumnRightPng,
      },
      EditTableEnum.deleteTable: {
        'value': 'Delete Table',
        'imagePath': ImageConstant.kiDeleteTablePng,
      },
    };

    value = typeMap[type]['value']!;
    imagePath = typeMap[type]['imagePath']!;
```